match/lib/match/encryption/encryption.rb

```diff
@@ -0,0 +1,148 @@
+require 'base64'
+require 'openssl'
+require 'securerandom'
+
+module Match
+  module Encryption
+    # This is to keep backwards compatibility with the old fastlane version which used the local openssl installation.
+    # The encryption parameters in this implementation reflect the old behavior which was the most common default value in those versions.
+    # As for decryption, 1.0.x OpenSSL and earlier versions use MD5, 1.1.0c and newer uses SHA256, we try both before giving an error
+    class EncryptionV1
+      ALGORITHM = 'aes-256-cbc'
+
+      def encrypt(data, password, salt, hash_algorithm = "MD5")
```

Comment:
Hmm, do we even still need to be able to do v1 encryption or could we just get rid of it. New files should just get v2 and old files can continue to use v1, then they will naturally expire over time and get replaced with v2

Reply:
Well, I use it in the tests :) I figured it didn't cause too much problem to keep it there as it brings some sort of symmetry to the code. We could also tag the class / methods as deprecated, but then I would have to update the tests as well.